### ISSUE TITLE
The examples currently inherit `parol_runtime` feature `trim_parse_tree`

### ISSUE THREAD

**Issue by jsinger67** [jsinger67/parol#58]
Since we incorporated different packages into one workspace, we can't useful handle different feature sets in different workspace members.
This is due to the fact that cargo builds crates finally with an union of all set features, see [Cargo reference](https://doc.rust-lang.org/cargo/reference/features.html#feature-unification) for details.
The `parol_runtime` feature `trim_parse_tree` is deliberately set in the crate `parol-ls` for performance reasons.
In contrast in examples, we actually want to see the full parse tree and performance is not a primary goal.

I'm not sure if a feature is a good choice to control if full parse trees are generated.

Maybe we should opt for a different configuration mechanism. For example, use a config file, an environment variable or a configuration method to choose which behavior to enable at runtime.

Unfortunately, this will in the end introduce an incompatibility when this feature is removed.

**Comment by jsinger67** [comment]
I'm fixing this issue by removing the feature `trim_parse_tree` from crate `parol_runtime`.
The feature will instead be activated by an equivalent function `trim_parse_tree` at the `parol::build::Builder` instance:
```rust
    Builder::with_explicit_output_dir("src")
        .grammar_file("oberon2.par")
        .expanded_grammar_output_file("../oberon2-exp.par")
        .parser_output_file("oberon2_parser.rs")
        .actions_output_file("oberon2_grammar_trait.rs")
        .enable_auto_generation()
        .user_type_name("Oberon2Grammar")
        .user_trait_module_name("oberon2_grammar")
        .trim_parse_tree()
        .generate_parser()
        .unwrap();
```
In any case, removing a feature from a crate is a very incompatible step.
I therefore apologise in advance.

I'm considering to change some internals about configuration. I think about extracting a common basis from `parol::build::Builder`, `parol::arguments::CliArgs`, `parol::generators::parser_generator::ParserData` and `parol::generators::user_trait_generator::UserTraitGenerator` into a separate structure.

Further I consider to provide a special switch at `parol new` subcommand to enable parse tree layout generation (and thereby disabling the trim_parse_tree behavior in newly created parol crates.

The work for this is done on branch *fix/feature_trim_parse_tree*.

**Comment by jsinger67** [comment]
Fixed with commit 320fa6ddbc4b8f23d726eef506258e693859b4b2
Use `parol` 0.18.1 and `parol_runtime` 0.14.0 to have this fix included.

Have fun! 🎉
